analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Tests/SymbolicValueAssertionsTest.cs

```diff
@@ -0,0 +1,160 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using SonarAnalyzer.SymbolicExecution.Roslyn;
+using SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution;
+
+namespace SonarAnalyzer.UnitTest.TestFramework.Tests
+{
+    [TestClass]
+    public class SymbolicValueAssertionsTest
+    {
+        [TestMethod]
+        public void SymbolicValue_HaveOnlyConstraint_Pass()
+        {
+            var sv = new SymbolicValue().WithConstraint(BoolConstraint.True);
+            sv.Invoking(x => x.Should().HaveOnlyConstraint(BoolConstraint.True)).Should().NotThrow();
+        }
+
+        [TestMethod]
+        public void SymbolicValue_HaveOnlyConstraint_WrongConstraint()
+        {
+            var sv = new SymbolicValue().WithConstraint(BoolConstraint.True);
+            sv.Invoking(x => x.Should().HaveOnlyConstraint(BoolConstraint.False, because: "{0} say so", "I")).Should()
+                .Throw<AssertFailedException>()
+                // * = x or SymbolicValue depending on compilation
```

Comment:
What does this mean? I'm getting `x` for both net7 and net48.
```suggestion
                // * = "x" or "SymbolicValue" depending on compilation
```

Reply:
It is in CI only and I didn't look at why it fails there:
https://dev.azure.com/sonarsource/DotNetTeam%20Project/_build/results?buildId=64307&view=ms.vss-test-web.build-test-results-tab